dds/DCPS/Serializer.h

```diff
@@ -652,6 +652,16 @@ class OpenDDS_Dcps_Export Serializer {
    */
   bool write_delimiter(size_t size);
 
+  enum ConstructionStatus {
+    ConstructionSuccessful,
+    ElementConstructionFailure,
+    BoundConstructionFailure
+  };
+
+  ConstructionStatus read_construction_status();
+
+  void write_construction_status(ConstructionStatus cs);
```

Comment:
It might be a good idea to rename these. The other functions in this file with "read" and "write" in their names are reading from or writing to the serialization stream. I think the convention in OpenDDS is to use the same name for both functions (`construction_status`), but I think `get_construction_status` and `set_construction_status` would also be acceptable if that suits you.